### PR TITLE
New feature added: add a node at a given position

### PR DESCRIPTION
**Pull request by sumitrai**
@hackerhorse please review it

### CODE REVIEW

File: lib/list.c [hunk 1]
```diff
@@ -48,3 +49,33 @@ void add_node_end(struct slist *head, struct slist *node)
...
+void add_node_pos(struct slist **head, struct slist *node, int insert_at_pos)
+{
+    int count;
+    struct slist *pos, *prev_pos;
+    struct slist *headp = *head;
+
+    pos = prev_pos = NULL;
+
+    // fix invalid user input
+    if (insert_at_pos <= 0)
```
Comment (prashinde) [inline lib/list.c:63]: This should be assert if it is invalid

File: lib/list.c [hunk 2]
```diff
@@ -48,3 +49,33 @@ void add_node_end(struct slist *head, struct slist *node)
...
+	insert_at_pos = 1;
+    
+    for (count = 1, pos = headp; pos != NULL; pos = pos->next, count++) {
```
Comment (HackerHorse) [inline lib/list.c]: Why not loop till "count == insert_at_pos"

for(count = 1, pos = headp; count != insert_at_pos; pos = pos->next, count++);

File: lib/list.c (hunk 1 above)
Comment (sumitrai) [inline lib/list.c:63]: Branching main again and fixing it (I merged upstream and delete the branch in the absent of review comments).

File: lib/list.c (hunk 2 above)
Comment (prashinde) [inline lib/list.c:66]: complicated! Make it simple and more readable

File: lib/list.c [hunk 3]
```diff
@@ -48,3 +49,33 @@ void add_node_end(struct slist *head, struct slist *node)
...
+	if (count == insert_at_pos) {
```
Comment (prashinde) [inline lib/list.c:67]: not pretty

File: lib/list.c [hunk 4]
```diff
@@ -48,3 +49,33 @@ void add_node_end(struct slist *head, struct slist *node)
...
+	    break;
+	}
+	prev_pos = pos;
+    }
+
+    // insert at postion 1: make node the new head
+    if (pos == headp) {
```
Comment (HackerHorse) [inline lib/list.c]: Linus Torvald's video

**Pull request merged**